### PR TITLE
Moved `force` parameter out of authorise() to AuthOptions

### PR DESCRIPTION
**Pull request by bladeSk**
This PR fixes https://github.com/ably/ably-php/issues/23

### CODE REVIEW

File: src/Auth.php [hunk 1]
```diff
@@ -93,17 +93,24 @@ public function isUsingBasicAuth() {
...
         if ( !empty( $tokenParams ) ) {
-            $this->defaultAuthoriseTokenParams = array_merge( $this->defaultAuthoriseTokenParams, $tokenParams );
+            $tokenParamsCopy = $tokenParams;
+            if ( isset( $tokenParamsCopy['timestamp'] ) ) unset( $tokenParamsCopy['timestamp'] );
+
+            $this->defaultAuthoriseTokenParams = array_merge( $this->defaultAuthoriseTokenParams, $tokenParamsCopy );
         }
         if ( !empty( $authOptions ) ) {
-            $this->defaultAuthoriseAuthOptions = array_merge( $this->defaultAuthoriseAuthOptions, $authOptions );
+            $authOptionsCopy = $authOptions;
```
Comment (mattheworiordan) [inline src/Auth.php]: I am not that fimiliar with PHP, but would this line simply create a new reference as opposed to create a duplicate of the associative array?  If so, that would mean that the following line `unset( $authOptionsCopy['force'] )` would in fact remove `force` from both.

**Comment by mattheworiordan** [comment]
Looks good, interested to get your thoughts on my comment though before merging

File: src/Auth.php (hunk 1 above)
Comment (bladeSk) [inline src/Auth.php]: In fact, no. PHP is totally backwards and arrays (but not objects!) are copied by value http://stackoverflow.com/a/1533214/388994 Hence the name authOptions**Copy**.

Comment (mattheworiordan) [inline src/Auth.php]: Ok, very backwards, I saw the name but expected that was a bug, hence the question.  Thanks for clarifying.

**Pull request merged**